### ISSUE TITLE
Wrong distance in the profile tool

### ISSUE THREAD

**Issue by saberraz** [qgis/QGIS#48270]
### What is the bug or the crash?

The distance (x axis) appears to be incorrect:

The identify tool displays the distance to be slightly less than 80km. But the same line appears to be over 100km in the profile tool along x axis. Ping me for the data and I can share it with you.
![image](https://user-images.githubusercontent.com/2649047/164026764-c5c5b5af-d61a-4516-b381-a3e9172494fa.png)

### Steps to reproduce the issue

1- Load a GPX dataset
2- Create a profile
3- Compare the length from identify tool vs. x axis.

### Versions

QGIS master

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

**Comment by nicogodet** [comment]
@saberraz project CRS ?
Looks similar to a current issue in plugin 😅

https://github.com/PANOimagen/profiletool/issues/65

**Comment by saberraz** [comment]
3857!

**Comment by nyalldawson** [comment]
This is not a trivial fix, if indeed it is a bug.

The profile tool is based entirely on planar calculations in the project's selected CRS, not ellipsoidal/great circle distances. I honestly doubt it's feasible at this stage to create a ellipsoidal aware profile tool -- the underlying libraries like libgeographic which we use for handling ellipsoidal calculations just don't have any capacity for handling elevations (I suspect because the mathematics required for these are horrendously complex!). 

Non-planar profiles would open a whole bunch of related questions too... eg if I draw a long enough profile line, would I expect to see the Earth's curvature shown in the profile chart? And if so, then you'd also expect to see vertical features drawn perpendicular to the surface from their base point. Or would the profile line always be considered as occurring along the surface of the ellipsoid?

Leaving everything as planar and requiring users to make a suitable local projection choice is an acceptable limitation in my view. 

We could consider adding a check and user facing warning here though, e.g. by measuring the profile line's ellipsoidal length and comparing against it's cartesian length, and if the difference exceeds say 2% then show a warning advising the user that their choice of CRS is misleading and to review this choice.

**Comment by saberraz** [comment]
I am easy with a warning. Another safer option would be to block users from creating the profile altogether if the difference is more than 2% and show a warning message.

**Comment by gacarrillor** [comment]
With the introduction of the `QgsProfileSourceRegistry` (see [this](https://github.com/qgis/QGIS/pull/57311)), this issue might need to be reconsidered.

We use distance/height calculations from the Swiss National Mapping Agency (Swisstopo) via web API.
Numbers indicate we get ellipsoidal distances. When visualizing the results on a project with EPSG:3857, we get significant distance deviations.

![image](https://github.com/qgis/QGIS/assets/652785/26fddce3-cedf-4323-8c9a-7d63347f5763)

@nyalldawson, could we override the profile canvas' max width with the max distance we get from the API (or with the max distance from all profiles sources available) to get rid of this issue?

See https://github.com/opengisch/qgis-swiss-locator/issues/86

**Comment by nyalldawson** [comment]
@gacarrillor 
> could we override the profile canvas' max width with the max distance we get from the API (or with the max distance from all profiles sources available) to get rid of this issue?

I'm -1 to that -- we'll just be providing obfuscated incorrect values if we go that route. This would still be my preferred approach:

> We could consider adding a check and user facing warning here though, e.g. by measuring the profile line's ellipsoidal length and comparing against it's cartesian length, and if the difference exceeds say 2% then show a warning advising the user that their choice of CRS is misleading and to review this choice.

**Comment by gacarrillor** [comment]
> This would still be my preferred approach:
> 
> > We could consider adding a check and user facing warning here though, e.g. by measuring the profile line's ellipsoidal length and comparing against it's cartesian length, and if the difference exceeds say 2% then show a warning advising the user that their choice of CRS is misleading and to review this choice.

Fair enough. We've switched to planar (2d) cartesian distances instead to stick to QGIS elevation profile functionality.
